listenbrainz/db/user_timeline_event.py

```diff
@@ -69,6 +69,33 @@ def create_user_track_recommendation_event(user_id: int, metadata: RecordingReco
         metadata=metadata,
     )
 
+def delete_user_timeline_event(user_id: int, metadata: RecordingRecommendationMetadata) -> UserTimelineEvent:
+    """ Deletes event from database and returns it.
+        Returns None if it couldn't find the queried event.
+    """
+    try:
+        meta = []
+        for i in metadata.dict().items():
+            if i[1] != None:
+                meta.append(" metadata ->> '" + i[0] + "' = '" + i[1] + "'")
```

Comment:
Don't construct the query by hand! This is a bad practice and makes the code vulnerable to SQL injection. A better way to do this is something like. https://github.com/metabrainz/listenbrainz-server/blob/b79dfb7194869939fe4a94c26cff8e8598b36bc8/listenbrainz/db/pinned_recording.py#L83-L89

Reply:
also I wanted to ask, does sqlalchemy provide sanitisation like this for JSONB queries?